optuna/integration/wandb.py

```diff
@@ -126,10 +156,25 @@ def __call__(self, study: optuna.study.Study, trial: optuna.trial.FrozenTrial) -
         metrics = {name: value for name, value in zip(names, trial.values)}
         attributes = {"direction": [d.name for d in study.directions]}
 
-        wandb.config.update(attributes)
-        wandb.log({**trial.params, **metrics}, step=trial.number)
+        run_id = trial._user_attrs.get("run_id")
+
+        if not (self._as_sweeps and run_id):
+            run = wandb.run or self._initialize_run()
+            run.log({**trial.params, **metrics})
+
+        else:
+            # If user is already logging wandb metrics from inside a trial
+            # reuse that run to log these study metrics
```

Comment:
We expect a comment is a full sentence (ref: https://github.com/optuna/optuna/wiki/Coding-Style-Conventions#block-and-inline-comments-are-supposed-to-be-sentences):
```suggestion
            # reuse that run to log these study metrics.
```